DOC: Mention both series and dataframe mirror methods

teoguso opened the pull request:
- [x] closes #21825

Now the automatically-generated doc features both `Series` and `Dataframe` in the _See also:_ section.

teoguso committed:
Mention both series and dataframe mirror methods

WillAyd commented on pandas/core/generic.py:
```diff
@@ -7847,6 +7847,8 @@ def _where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         --------
         :func:`DataFrame.%(name_other)s` : Return an object of same shape as
```
Can you change `self` to `caller` both here and the line below?

teoguso replied:
Absolutely. I also modified the notes, so that "caller" is used instead of "DataFrame" (made little sense in the context of Series). I hope that makes sense.

teoguso committed:
Use "caller" instead of self / Dataframe

In the notes, a dataframe was used as example,
which would not make sense when the documentation
appies to a series.
Now the nomenclature "caller" and "other" is used,
so that it works for both series and dataframe.

WillAyd commented on pandas/core/generic.py:
```diff
@@ -7832,12 +7832,13 @@ def _where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
 
-        The signature for :func:`DataFrame.where` differs from
-        :func:`numpy.where`. Roughly ``df1.where(m, df2)`` is equivalent to
+        The signature for :func:`DataFrame.where` or
```
Think you meant to say `Series` for one of these

teoguso replied:
The problem is that this document will be used both for Series and for DataFrame. Therefore that paragraph would be odd in at least one of the two cases. I'm really not sure how to solve this.

WillAyd replied:
Just say `Series` for one and `DataFrame` for the other. Right now you are saying `DataFrame` for both

teoguso replied:
I'm not sure what you mean there... There is one piece of documentation for both, how can I differentiate?

WillAyd replied:
My point is you don't need to differentiate (I mean you could, but its a lot more work and I don't think its worth the effort).

Right now you are saying "The signature for `DataFrame.where` or `DataFrame.where` differs from `numpy.where`". Just change that to "The signature for `Series.where` or `DataFrame.where` ..."

teoguso replied:
I see! Sorry, it's just that I was hoping for a more automated solution. :-D

WillAyd replied:
There's probably a way to add substitution for say a `klass` keyword argument (we do this in other instances). Welcome to give it a shot if you like!

teoguso replied:
I'll see what I can do!

WillAyd reviewed (changes requested):
minor comment otherwise lgtm

jreback commented:
@datapythonista if you have any comments.

datapythonista committed:
Fixing typo (DataFrame instead of Series), and minor formatting.

pep8speaks commented:
Hello @teoguso! Thanks for updating the PR.

 - There are no PEP8 issues in the file [`pandas/core/generic.py`](https://github.com/teoguso/pandas/blob/cd15aa4837ad97c4a61cd8441a3ad6fbc29f615b/pandas/core/generic.py) !

datapythonista reviewed (approved):
@WillAyd we'll have to find an easy way to display `Series` for the `Series` docstring, and `DataFrame` for the `DataFrame` one at some point. `%(klass)s` renders as `NDFrame`. But I'd merge this PR as it is, and take care of that later on when we find a way.

Please merge if the changes look ok to you.

jreback commented:
closing as stale. if you'd like to continue, pls ping.

datapythonista commented:
@jreback this is not stale, it's ready to be merged

datapythonista commented:
@teoguso can you please merge master and fix the conflicts?

datapythonista commented:
@teoguso do you have time to fix the conflicts, so we can merge?

jreback commented:
@datapythonista
